server/routers/api-router.js

```diff
@@ -87,6 +88,7 @@ router.get("/api/push/:pushToken", async (request, response) => {
 
         io.to(monitor.user_id).emit("heartbeat", bean.toJSON());
         Monitor.sendStats(io, monitor.id, monitor.user_id);
+        new Prometheus(monitor).update(bean, undefined);
```

Comment:
Since I don't use Prometheus, may need someone to cross check.

As here `new Prometheus(monitor)` is created a new Object for each request, I am not sure it is really ok.

Reply:
This is the correct way to use the current design. Since we don't have a list of existing metrics, only way to update is to construct the object with the same parameters. It's a very simple object with some identifying fields.

Reply:
Sorry to jump in.
But this would be solved with #898, where I'm currently working on and my current suggested commit (https://github.com/louislam/uptime-kuma/commit/054b3ec1c5c0369bc9ab4f95b2f02d27cae96c62) for the PR would allow access to the existing Prometheus instance of the monitor on `monitor.prometheus.update(...)`.

BTW: For the notes of @louislam 😉 We could think about to `monitor.prometheus.update(...)` in `onAfterUpdate` on the bean... I don't know `redbean-node` good enough, but it sounds like the monitor could implement everything that has to be done on update there. So every `store` would automatically update prometheus if required? This would hide monitor internal stuff from api-router.

Reply:
you can now replace this line with `monitor.getPrometheus().update(bean, undefined);`